### PR TITLE
Add config watchers

### LINKED ISSUE #74: SampleSource should use the new config watcher
This enforces well configured metrics/tracing/logging for new sources.

Example https://github.com/knative/eventing-contrib/pull/1215

### PR DESCRIPTION
**Pull request by slinkydeveloper**
Fixes #74

I've also modified the update-deps script to pull latest eventing version from master (like in eventing-contrib)

/assign @lberk

### CODE REVIEW

**Comment by slinkydeveloper** [comment]
ok i've found that there is a _not so_ funny data race in sdk-go, next release will be out soon with the fix

**Comment by slinkydeveloper** [comment]
Ok now it should work fine :smile:

File: pkg/reconciler/sample/controller.go [hunk 1]
```diff
@@ -51,6 +53,8 @@ func NewController(
 	r := &Reconciler{
 		dr:  &reconciler.DeploymentReconciler{KubeClientSet: kubeclient.Get(ctx)},
 		sbr: &reconciler.SinkBindingReconciler{EventingClientSet: eventingclient.Get(ctx)},
+		// Config accessor takes care of tracing/config/logging config propagation to the receive adapter
+		configAccessor: reconcilersource.WatchConfigurations(ctx, "sample-source", cmw),
```
Comment (lberk) [inline pkg/reconciler/sample/controller.go:57]: @slinkydeveloper just for my own understanding, what specifically does the 'component' parameter actually point to?

Comment (slinkydeveloper) [reply pkg/reconciler/sample/controller.go:57]: it's a name used for bootstrapping the metrics exporter, it represents the name of the source

**Pull request merged**